File: file.go
package main

import (
	"fmt"
	"strconv"
)

type Person struct {
	firstName string
	lastName  string
	birthYear int
}

func (p *Person) uniqueId() string {
	return string(p.firstName[0]) + p.lastName + strconv.Itoa(p.birthYear)
}

func main() {
	gary := Person{"Gary", "Lime", 1982}

	id := gary.uniqueId()

	fmt.Println(id)
}

File: file.ts
class Person {
  constructor(
    private firstName: string,
    private lastName: string,
    private birthYear: number
  ) {}

  uniqueId(): string {
    return this.firstName[0] + this.lastName + this.birthYear;
  }
}

const gary = new Person("Gary", "Lime", 1982);

const id = gary.uniqueId();

console.log(id);
